bbj-vscode/src/language/java-interop.ts

```diff
@@ -142,16 +146,24 @@ export class JavaInteropService {
             }
             for (const method of javaClass.methods) {
                 (method as Mutable<JavaMethod>).$type = JavaMethod;
+                const methodDocs = isClassDoc(documentation) ? documentation.methods.filter(
+                    m => m.name == method.name
+                        && m.params.length === method.parameters.length
+                ) : [];
                 method.resolvedReturnType = {
                     ref: await this.resolveClassByName(method.returnType, token),
                     $refText: method.returnType
                 };
-                for (const parameter of method.parameters) {
+                for (const [index, parameter] of method.parameters.entries()) {
                     (parameter as Mutable<JavaMethodParameter>).$type = JavaMethodParameter;
                     parameter.resolvedType = {
                         ref: await this.resolveClassByName(parameter.type, token),
                         $refText: parameter.type
                     };
+                    if (methodDocs.length > 0) {
+                        // TODO check types of parameters
+                        parameter.realName = methodDocs[0].params[index]?.name
```

Comment:
I would like to keep this names separately, this also makes clear that only some of the JavaClasses will become a "real" source name.

> for the case when the name is of pattern /^arg\d+$/i?

One can imaging that some method has a parameter named `arg0` in the source code, so you can no longer distinguish between binary name and source name.

> Alternatively I would have proposed a helper function to get the actual name.

The `realName` property is only used in two places, so I don't see a benefit to create a utility yet.

Reply:
Yes, it sounds like too much, but will you or me or someone else remember this change in a few months? What if more of those `realName ?? name` occur or you need to make a change and do not get all occurrences?

If the `realName` can be the same as the `name` I see no problem in using only `name`. Binary name is only relevant if you have no source name.
What would be the benefit of having both names in the object?

Reply:
> What if more of those realName ?? name occur

Then we can think about a utility :) 

> If the realName can be the same as the name I see no problem in using only name. Binary name is only relevant if you have no source name.

No, I don't think it can be mixed. `name` is what we become from the Java backend and an **optional** `realName` property is enhancing the `JavaClass` AST type to cache the value we get (but mostly not) from reading the Json files.
We should really not mix two different things coming from different sources into one property. 

> What would be the benefit of having both names in the object? 

Separated concepts, makes the data structure and flow understandable.